src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -50,30 +66,88 @@ export default class SqlX extends SelfServeBaseClass {
   };
 
   public onSave = async (currentValues: Map<string, SmartUiInput>): Promise<SelfServeNotification> => {
-    validate(currentValues);
-    // TODO: add pre processing logic before calling the updateDedicatedGatewayProvisioning() RP call.
-    throw new Error(`onSave not implemented. No. of properties to save: ${currentValues.size}`);
+    const response = await getCurrentProvisioningState();
+
+    // null implies the resource has not been provisioned.
+    if (response.status !== undefined && response.status !== ResourceStatus.Running.toString()) {
+      switch (response.status) {
+        case ResourceStatus.Creating.toString():
+          return { message: "CreateMessage", type: SelfServeNotificationType.error };
+        case ResourceStatus.Updating.toString():
+          return { message: "UpdateMessage", type: SelfServeNotificationType.error };
+        case ResourceStatus.Deleting.toString():
+          return { message: "DeleteMessage", type: SelfServeNotificationType.error };
```

Comment:
Add different messages here.
Let refresh messages be something like "Updation ongoing"
Let failure messages be something like "Delete failed since updation is ongoing"